src/main/java/Duke.java

```diff
@@ -1,10 +1,63 @@
+import duke.Storage;
+import duke.task.TaskList;
+import duke.Ui;
+import duke.DukeException;
+import duke.command.Parser;
+import duke.command.Command;
+
+/**
+ * The type Duke.
+ */
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Instantiates a new Duke.
+     *
+     * @param filePath the file path
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Run.
+     */
+    public void run() {
```

Comment:
Could possibly use protected here instead of public

Reply:
True. Thanks for the suggestion. However, I was considering the case where a user might want to call run() again on an already created Duke object.